code.ts

```diff
@@ -139,46 +151,69 @@ function secondsToInterval(seconds: number): string {
   return result;
 }
 
+
+/** 
+* Code that updates all timers on the Figma stage
+* will also send updates / messages to UI.html, so we can show timers there
+*/
+
 async function startTimer(node: TextNode, seconds: number, template: string, startsWithTimer: boolean) {
   await figma.loadFontAsync(node.fontName as FontName);
-  activeTimer += 1;
+  totalTimers += 1
 
   console.log("Timer started / became active");
 
-  var timerID = activeTimer;
+  var timerID = totalTimers;
   var keepItRunning = true;
   var secondsToGo = seconds;
   var newText = "";
 
+  adjustUIWindowHeight();
+  figma.ui.postMessage(["start timer", newText, timerID, secondsToGo, seconds]);
+
+  // this loop updates all timers every second
   while (keepItRunning) {
 
     // checking if reset was clicked by user and if so resetting all timers
     if (reset) {
-      secondsToGo = seconds;
-      newText = fillUpTimeStringWithTemplate(secondsToInterval(secondsToGo), template);
-      if (startsWithTimer) {
-        newText = "Timer: " + newText;
-      }
-      node.characters = newText;
+      newText = fillUpTimeStringWithTemplate(secondsToInterval(seconds), template);
+      //updateTimerText(startsWithTimer, newText, node);
       keepItRunning = false;
     };
 
     // checking if pause was NOT clicked
-    if (!pause) {
-      if (secondsToGo > 0) {
-        newText = fillUpTimeStringWithTemplate(secondsToInterval(secondsToGo), template);
-        if (startsWithTimer) {
-          newText = "Timer: " + newText;
+    if (!reset) {
```

Comment:
We could use else here instead of !reset

Reply:
done